packages/timeline-state-resolver/src/integrations/tricaster/index.ts

```diff
@@ -0,0 +1,176 @@
+
+import { TimelineState } from 'superfly-timeline'
+import { DeviceType, Mappings, TriCasterOptions, DeviceOptionsTriCaster } from 'timeline-state-resolver-types'
+import { TriCasterState, TriCasterStateDiffer } from './triCasterStateDiffer'
+import { TriCasterCommandWithContext } from './triCasterCommands'
+import { TriCasterConnection } from './triCasterConnection'
+
+const DEFAULT_PORT = 5951
+
+export type DeviceOptionsTriCasterInternal = DeviceOptionsTriCaster
+
+export class TriCasterDevice extends DeviceWithState<TriCasterState, DeviceOptionsTriCasterInternal> {
+	private _doOnTime: DoOnTime
+
+	private _resolveInitPromise: (value: boolean) => void
+	private _connected = false
+	private _initialized = false
+	private _connection?: TriCasterConnection
+	private _stateDiffer?: TriCasterStateDiffer
+
+	constructor(deviceId: string, deviceOptions: DeviceOptionsTriCasterInternal, getCurrentTime: () => Promise<number>) {
+		super(deviceId, deviceOptions, getCurrentTime)
+
+		this._doOnTime = new DoOnTime(() => this.getCurrentTime(), SendMode.BURST, this._deviceOptions)
+		this._doOnTime.on('error', (e) => this.emit('error', 'TriCasterDevice.doOnTime', e))
+		this._doOnTime.on('slowCommand', (msg) => this.emit('slowCommand', this.deviceName + ': ' + msg))
+		this._doOnTime.on('slowSentCommand', (info) => this.emit('slowSentCommand', info))
+		this._doOnTime.on('slowFulfilledCommand', (info) => this.emit('slowFulfilledCommand', info))
+	}
+	async init(options: TriCasterOptions): Promise<boolean> {
+		const initPromise = new Promise<boolean>((resolve) => {
+			this._resolveInitPromise = resolve
+		})
+		this._connection = new TriCasterConnection(options.host, options.port ?? DEFAULT_PORT)
+		this._connection.on('connected', (info, shortcutStateXml) => {
+			this._stateDiffer = new TriCasterStateDiffer(info)
+			this._setInitialState(shortcutStateXml)
+			this._setConnected(true)
+			this._initialized = true
+			this._resolveInitPromise(true)
+			this.emit('info', `Connected to TriCaster ${info.productModel}, session: ${info.sessionName}`)
+		})
+		this._connection.on('disconnected', (_reason) => {
+			this._setConnected(false)
+		})
+		this._connection.on('error', (reason) => {
+			this.emit('error', 'TriCasterConnection', reason)
+		})
+		this._connection.connect()
+		return initPromise
+	}
+
+	private _setInitialState(shortcutStateXml: string): void {
+		if (!this._stateDiffer) {
+			throw new Error('State Differ not available')
+		}
+		const time = this.getCurrentTime()
+		const state = this._stateDiffer.shortcutStateConverter.getTriCasterStateFromShortcutState(shortcutStateXml)
+		this.setState(state, time)
+	}
+
+	private _connectionChanged(): void {
+		this.emit('connectionChanged', this.getStatus())
+	}
+
+	private _setConnected(connected: boolean): void {
+		if (this._connected !== connected) {
+			this._connected = connected
+			this._connectionChanged()
+		}
+	}
+
+	/** Called by the Conductor a bit before handleState is called */
```

Comment:
```suggestion
	/** Called by the Conductor shortly before handleState is called */
```

Just a more gramitically correct sentence :)

Also, is it suppose to say 'Conductor' and not 'Constructor'?

Reply:
Same story as here: https://github.com/tv2/tv-automation-state-timeline-resolver/pull/46#discussion_r1085058437